samples/golang/webfrontend/src/app/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+)
+
+// ServerMessage is a representation of a message from server
+type ServerMessage struct {
+	Value string
+}
+
+// PageVariables is a representation of a server page
+type PageVariables struct {
+	PageTitle   string
+	PageMessage ServerMessage
+	PageButton  string
+}
+
+var counter int
```

Comment:
You  might want to craft this incrementing counter as a goroutine with a channel that streams incrementing numbers back. It's a preferable option in Go to use channels than to resort to mutexes.